ffi_derive/src/lib.rs

```diff
@@ -132,6 +133,73 @@
 //!
 //! See `../tests/remote_types` for an example.
 //!
+//! Similarly, sometimes a type we want to expose will reference a typealias defined in a remote
+//! crate. We support that, but because the type information that backs the alias isn't available at
+//! the time procedural macros run, we require some additional configuration in both the module that
+//! defines the alias, and on the type whose fields reference the alias.
+//!
+//! ### Remote alias definitions
+//!
+//! When a module defines aliases that may be referenced by a type that derives an FFI, the
+//! `alias_resolution` attribute macro needs to be run on it in order to populate the definitions of
+//! those aliases somewhere so that we can reference them when resolving the underlying types of
+//! alias references. The macro invocation also needs to define a unique string for the module
+//! (which we refer to internally as the `resolution_key`). This will be used with a helper
+//! attribute on types that derive an FFI so that we can identify the source where their aliases are
+//! defined.
+//!
+//! Invoking the alias resolution macro on a module looks like this:
+//! ```
+//! #[ffi_derive::alias_resolution(some_unique_string)]
+//! mod aliases_here {
+//!     pub type Foo = u8;
+//! }
+//! ```
+//!
+//! Finally, an alias may be defined over another alias (which is odd but happens). We support those
+//! cases, but require an additional helper attribute on the alias declaration to tell us where
+//! *that* alias is defined. For example:
+//! ```ignore
+//! #[ffi_derive::alias_resolution(crate1_aliases)]
+//! mod aliases_in_crate1 {
+//!     pub type Foo = u8;
+//! }
+//!
+//! #[ffi_derive::alias_resolution(crate2_aliases)]
+//! mod aliases_in_crate2 {
+//!     #[nested_alias="crate1_aliases"]
+//!     pub type Bar = Foo;
+//! }
+//! ```
+//!
+//! ### Remote alias references
+//!
+//! When a type derives an FFI includes a field whose type is an alias defined in a remote crate,
```

Comment:
I'm feeling dense right now, but am having a hard time reading this... should it be "When a type-derived FFI includes..."?